src/windows/wslaservice/inc/wslaservice.idl

```diff
@@ -102,12 +109,93 @@ struct _VIRTUAL_MACHINE_SETTINGS {
 } VIRTUAL_MACHINE_SETTINGS;
 
 
-typedef
-struct _WSLA_SESSION_SETTINGS {
+
+struct WSLA_SESSION_SETTINGS {
     LPCWSTR DisplayName;
-    // Details TBD.
-} WSLA_SESSION_SETTINGS;
+    LPCWSTR StoragePath;
+
+    // TODO: Termination callback, flags
+};
+
 
+struct WSLA_REGISTRY_AUTHENTICATION_INFORMATION
+{
+    int Dummy; // Dummy value for the .idl file to compile.
+    // TODO: Details TBD.
+};
+
+
+struct WSLA_IMAGE_INFORMATION
+{
+    LPWSTR Name;
+    LPWSTR Hash;
+    ULONGLONG Size;
+    ULONGLONG DownloadTimestamp;
+};
+
+struct WSLA_PROCESS_OPTIONS
+{
+    LPCSTR Executable;
+    [unique] LPCSTR CurrentDirectory;
+    [size_is(CommandLineCount)] LPCSTR* CommandLine;
+    ULONG CommandLineCount;
+    [unique, size_is(EnvironmentCount)] LPCSTR* Environment;
+    ULONG EnvironmentCount;
+    [unique, size_is(EnvironmentCount)] WSLA_PROCESS_FD *Fds;
+    int FdsCount;
```

Comment:
What are these used for?

Reply:
These are the process' file descriptors, which can be handles, or linux files. 

Those are used for interactive and non interactive processes, and can be used to easily copy files from and to a container. For instance you can do `cat` with `stdin=/path/to/file` to easily get a file out of a container